api/index.js

```diff
@@ -1,89 +1,100 @@
+
+  request.credentials = 'include'
 
-  // Process body
-  if (!['GET', 'HEAD'].includes(method.toUpperCase())) {
-    if (typeof data == 'object') {
-      requestOptions.body = JSON.stringify(data)
-      requestOptions.headers['Content-Type'] = 'application/json'
-    } else requestOptions.body = data
+  return { ...init, ...request }
+}
+
+const processStatus = response => {
+  if (response.status >= 400) {
+    const Error = NetworkError.getErrorFromStatusCode(response.status)
+    throw new Error(
+      `Request for ${response.url} failed on ${response.status}`,
+      response
+    )
   }
 
-  // Process custom headers
-  Object.assign(requestOptions.headers, customHeaders)
-
-  // Make abortable
-  const controller = new AbortController()
-  const { signal } = controller
-  requestOptions.signal = signal
-
-  const fetchPromise = fetch(requestUrl, requestOptions)
-    .then(response => {
-      const { headers } = response
-      const type = headers.get('Content-Type')
-      const result = { type, headers }
-
-      if (response.status >= 400) {
-        const Error = NetworkError.getErrorFromStatusCode(response.status)
-        throw new Error(
-          `Request for ${method} ${url} failed on ${response.status}`,
-          response
-        )
-      }
-
-      if (method.toUpperCase() === 'HEAD') return result
-
-      const dataPromise = /application\/([\w.-]\+)?json/g.test(type)
-        ? response.json()
-        : response.blob()
-
-      return new Promise((resolve, reject) =>
-        dataPromise.then(
-          value =>
-            resolve({
-              data: value,
-              type,
-              headers,
-            }),
-          reason => reject(reason)
-        )
-      )
-    })
-    .catch(async error => {
-      const { response, message } = error
-      if (error instanceof NetworkError) throw error
-
-      // user is probably offline or CORS failed
-      throw new NetworkError(
-        `Request for ${method} ${url} failed. Response: ${response.status}, ${message}`
-      )
-    })
-
-  return new CancelablePromise(fetchPromise, {
-    cancel: () => controller.abort(),
-  })
+  return response
+}
+
+const processBody = response => {
+  const { headers } = response
+  const type = headers.get('Content-Type')
+
+  return (/application\/([\w.-]\+)?json/g.test(type)
+    ? response.json()
+    : response.blob()
+  ).then(data => ({ data, type, headers }))
+}
+
+// TODO: Remove later
+// Old type of error processing
+// eslint-disable-next-line no-unused-vars
+const processError = error => {
+  const { response, message } = error
+  if (error instanceof NetworkError) throw error
+
+  // user is probably offline or CORS failed
+  throw new NetworkError(
+    `Request for ${response?.url} failed. Response: ${response?.status}, ${message}`
+  )
+}
+
+const executeRequest = ({ url, ...options }) => {
+  fetch(url, options)
+    .then(processStatus)
+    .then(processBody)
```

Comment:
I think you are missing here `.catch(processError)`

Reply:
I do it for a reason. I cannot and do not want to pass `request` object to error processing. So if I remove the second part (converting any error to the `NewtorkError`), there is no need in it at all.

However, you put it there for a reason. If you insist I will put it back. Why did you convert any error to `NetworkError`?

Reply:
No, not really. There was no reason I just did it because I wanted to have a unified error. Feel free to remove it. You can remove [this property](https://github.com/oacore/dashboard/blob/api/api/errors.js#L6) too. It's not used.

Reply:
There is no need in removing the property.

However, there is a need in removing the `catch` clause because it can catch fetch initialisation/execution errors which may not be reported.

Reply:
For me, all unused code can (should be) removed and that property is not used anywhere. 😄